### DISCUSSION TITLE
v0.18.0

### DISCUSSION THREAD

**Discussion by afeiszli** [gravitl/netmaker#2032] [Announcements]
# ATTENTION: Do not attempt to upgrade to 0.18.0. This is for testing purposes only, and will remain in pre-release. Upgrading from a prior version will not succeed. You are welcome to try a fresh install of 0.18.0 for testing purposes, but do not run in production.

### What's New
- All New Netclient 
  - https://github.com/gravitl/netclient 
  - Apache 2.0 License
  - Proxy for STUN behind NAT
  - Operates on a single network interface
  - New GUI
  - Automatic client upgrades
    - clients will now track server version
- All New Hosts Functionality
  - Hosts represent machines
  - Nodes represent machines on networks
  - 1 Host --> Many Nodes
  - Hosts can be added to networks via UI 
  - Hosts can be made "Default Hosts"
    - Automatically added to any network
    - Relay functionality moved to host level
  - Removed Server Node
  - "Default Host" replaces Server Node functionality
  - Server no longer requires root, wireguard, or special networking permissions
- STUN server on Netmaker
- Ingress and Egress routing now operate without system commands
- Postup/Postdown removed
- Simplified Message Queue
- Removed "Point to Site"
- Health Check endpoint added to server
- Windows installer improved (does not require uninstall)

### What's Fixed
- Database Synchronization
- Node expiration works again (set expiration, node deletes)

### Known Bugs
- Upgrading to 0.18.0 WILL NOT WORK. Do not attempt it
- If a host becomes a zombie, you cannot delete it
- Local network does not work, will be depricated
- Interface IP disappears sometimes
- Only iptables nodes will work as ingress/egress
- Can't ping ext clients on one Ingress from an ext client on another ingress
- Failover (EE) does not work
- Internet gateway will not work

<hr /><em>This discussion was created from the release <a href='https://github.com/gravitl/netmaker/releases/tag/v0.18.0'>v0.18.0</a>.</em>

**Comment by GaZaTu** [comment]
Idk if this is the right place to ask but i read the `Postup/Postdown removed` bullet point and I'm a bit confused.
Is it gone completely or only in the server UI?
Because I actually need the feature to assign a `Postup` script on the client side to configure my firewall (firewalld on linux).

**Reply by mattkasun** [reply]
netclient no longer uses wg-quick up/down to control the wireguard interface(s) and as such the post-up/post-down fields no longer exist.

**Comment by miyu4u** [comment]
For a while I thought development had stopped due to the lack of commis on master, but I'm glad to see the version is up.

Any idea when i can expect to see a full release of 0.18.x?
